Sources/tart/Fetcher.swift

```diff
@@ -2,45 +2,35 @@ import Foundation
 import AsyncAlgorithms
 
 class Fetcher: NSObject, URLSessionTaskDelegate, URLSessionDelegate, URLSessionDataDelegate {
-    let responseCh = AsyncThrowingChannel<URLResponse, Error>()
+  func fetch(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, HTTPURLResponse) {
     let dataCh = AsyncThrowingChannel<Data, Error>()
 
-    func fetch(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, URLResponse) {
-        let task = URLSession.shared.dataTask(with: request)
-        task.delegate = self
-        task.resume()
+    let (data, response) = try await URLSession.shared.data(for: request)
 
-        // Wait for the response and only then return
-        var iter = responseCh.makeAsyncIterator()
-        let response = try await iter.next()!
+    Task {
+      await dataCh.send(data)
 
-        return (dataCh, response)
+      dataCh.finish()
     }
 
-    func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive response: URLResponse) async -> URLSession.ResponseDisposition {
-        await responseCh.send(response)
+    return (dataCh, response as! HTTPURLResponse)
+  }
 
-        return .allow
-    }
+  func fetchViaFile(_ request: URLRequest) async throws -> (AsyncThrowingChannel<Data, Error>, HTTPURLResponse) {
+    let dataCh = AsyncThrowingChannel<Data, Error>()
 
-    func urlSession(_ session: URLSession, dataTask: URLSessionDataTask, didReceive data: Data) {
-        let sema = DispatchSemaphore(value: 0)
+    let (fileURL, response) = try await URLSession.shared.download(for: request)
+    let fh = try FileHandle(forReadingFrom: fileURL)
+    try FileManager.default.removeItem(at: fileURL)
```

Comment:
Let's add a comment explaining the magic of acquiring a file handle and then removing the file but still being able to read it.

Reply:
See https://github.com/cirruslabs/tart/pull/306/commits/9363c7cfd7131d8a51fd02c2de96f37984c5246d.